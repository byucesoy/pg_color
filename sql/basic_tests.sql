CREATE TABLE colors(color color);

-- INSERT different color values
INSERT INTO colors VALUES('000000'), ('FFFFFF'), ('FF0000'), ('00FF00'), ('0000FF');

-- Display INSERTed values
SELECT * FROM colors;

-- Try INSERTing invalid colors
INSERT INTO colors VALUES('0');
INSERT INTO colors VALUES('0000000');
INSERT INTO colors VALUES('GGGGGG');

-- Test equality operator
SELECT * FROM colors WHERE color = '000000';

-- Test inequality operator
SELECT * FROM colors WHERE color <> '000000';

-- Test JOIN with our new type
SELECT * FROM colors c1 JOIN colors c2 ON (c1.color = c2.color);
