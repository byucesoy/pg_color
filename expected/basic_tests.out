CREATE TABLE colors(color color);
-- INSERT different color values
INSERT INTO colors VALUES('000000'), ('FFFFFF'), ('FF0000'), ('00FF00'), ('0000FF');
-- Display INSERTed values
SELECT * FROM colors;
 color  
--------
 000000
 FFFFFF
 FF0000
 00FF00
 0000FF
(5 rows)

-- Try INSERTing invalid colors
INSERT INTO colors VALUES('0');
ERROR:  value "0" is out of range for type color
LINE 1: INSERT INTO colors VALUES('0');
                                  ^
INSERT INTO colors VALUES('0000000');
ERROR:  value "0000000" is out of range for type color
LINE 1: INSERT INTO colors VALUES('0000000');
                                  ^
INSERT INTO colors VALUES('GGGGGG');
ERROR:  value 'G' is not a valid digit for type color.
LINE 1: INSERT INTO colors VALUES('GGGGGG');
                                  ^
-- Test equality operator
SELECT * FROM colors WHERE color = '000000';
 color  
--------
 000000
(1 row)

-- Test inequality operator
SELECT * FROM colors WHERE color <> '000000';
 color  
--------
 FFFFFF
 FF0000
 00FF00
 0000FF
(4 rows)

-- Test JOIN with our new type
SELECT * FROM colors c1 JOIN colors c2 ON (c1.color = c2.color);
 color  | color  
--------+--------
 000000 | 000000
 FFFFFF | FFFFFF
 FF0000 | FF0000
 00FF00 | 00FF00
 0000FF | 0000FF
(5 rows)

